system/HTTP/URI.php

```diff
@@ -593,8 +593,20 @@ public function getTotalSegments(): int
 	 */
 	public function __toString(): string
 	{
+		// If hosted in a sub-folder, we will have additional
+		// segments that show up prior to the URI path we just
+		// grabbed from the request, so add it on if necessary.
+		$baseUri  = new URI(config(\Config\App::class)->baseURL);
```

Comment:
should can use `new self(...)`:

```suggestion
              $baseUri  = new self(config(\Config\App::class)->baseURL);
```

Reply:
Thanks!